src/Symfony/Component/Form/Extension/Csrf/Type/FormTypeCsrfExtension.php

```diff
@@ -70,7 +76,7 @@ public function finishView(FormView $view, FormInterface $form, array $options):
     {
         if ($options['csrf_protection'] && !$view->parent && $options['compound']) {
             $factory = $form->getConfig()->getFormFactory();
-            $tokenId = $options['csrf_token_id'] ?: ($form->getName() ?: $form->getConfig()->getType()->getInnerType()::class);
+            $tokenId = $options['csrf_token_id'] ?: $form->getConfig()->getAttribute('csrf_token_id');
```

Comment:
With the updated code above can it actually happen that `$options['csrf_token_id']` isn't set here?

Reply:
yep: $options modifications are not visible from here, that's why we need to use attributes

Reply:
what about moving `$builder->setAttribute('csrf_token_id', $options['csrf_token_id'])` outside/after the `if` above and relying on `$form->getConfig()->getAttribute('csrf_token_id')` as the last source of truth? that will also be simpler for externals to get the used `csrf_token_id` instead of checking the form options then the form attribute